<a href="https://colab.research.google.com/github/AnnaMirovna/A-B-testing/blob/main/%D0%98%D1%82%D0%BE%D0%B3%D0%BE%D0%B2%D0%B0%D1%8F_%D0%B0%D1%82%D1%82%D0%B5%D1%81%D1%82%D0%B0%D1%86%D0%B8%D1%8F_%D0%90%D0%91_%D1%82%D0%B5%D1%81%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Задача 1.
Подведите результаты эксперимента в экселе по следующим данным: ab_stats.csv - Google Диск:
Стат значимо ли отличается ARPPU в двух группах ? Какие рекомендации дадите
менеджеру?

In [8]:

!pip install pandas scipy

import pandas as pd
from scipy.stats import ttest_ind
from google.colab import files

data = pd.read_csv('ab_stats(1).csv')

# Шаг 1: Фильтрация пользователей, которые совершили покупки (num_purchases > 0)
data_paying_users = data[data['num_purchases'] > 0].copy()

# Шаг 2: Расчет ARPPU (Average Revenue Per Paying User)
data_paying_users['ARPPU'] = data_paying_users['revenue'] / data_paying_users['num_purchases']

# Шаг 3: Вычисление среднего значения ARPPU для каждой группы
arppu_grouped = data_paying_users.groupby('ab_group')['ARPPU'].mean()

# Шаг 4: Проведение t-теста
arppu_A = data_paying_users[data_paying_users['ab_group'] == 'A']['ARPPU']
arppu_B = data_paying_users[data_paying_users['ab_group'] == 'B']['ARPPU']

# t-тест
t_stat, p_value = ttest_ind(arppu_A, arppu_B)

# Шаг 5: Вывод результатов
print("Среднее значение ARPPU по группам:")
print(arppu_grouped)
print("\nРезультаты t-теста:")
print(f"t-статистика: {t_stat}")
print(f"p-значение: {p_value}")


Среднее значение ARPPU по группам:
ab_group
A    6.548581
B    6.560296
Name: ARPPU, dtype: float64

Результаты t-теста:
t-статистика: -0.012886870841395809
p-значение: 0.9897233159320268


Средние значения ARPPU (средняя выручка на одного покупателя) в двух группах:

  Группа A: 6.55
  Группа B: 6.56

Результаты t-теста:

  t-статистика: -0.013
  p-значение: 0.99

Вывод:

Различие в ARPPU между группами A и B статистически незначимо (p-значение > 0.05). Это означает, что нет достаточных оснований утверждать, что одна из групп показывает лучшее финансовое поведение по сравнению с другой.

Рекомендации для менеджера:
Поскольку различия между группами незначительны, текущие изменения, вероятно, не оказывают влияния на ключевой показатель ARPPU.
Возможно, стоит пересмотреть экспериментальные условия или попробовать другие метрики (например, LTV, конверсию), чтобы выявить значимые различия.
Если важны незначительные улучшения, можно продолжить тестирование, увеличив выборку, чтобы убедиться, что результаты устойчивы.

# Задача 4.
Вы решили сравнивать метрику CPA в двух группах. Размер выборки - 2350 элементов в
каждой группе.
Для проверки нормальности распределения на выборке в 2350 наблюдений применили ,
критерий Шапиро-Уилка и получили p-value, равный 0.00002, alpha = 5%.
Какой бы вывод мы могли сделать в данном случае?
В этом случае какой статистический критерий для проверки первоначальной гипотезы тут лучше
всего подойдёт и почему?

1. Вывод по критерию Шапиро-Уилка

Критерий Шапиро-Уилка используется для проверки нормальности распределения данных. Он возвращает p-value, и интерпретация его заключается в следующем:

  Нулевая гипотеза (H0): Данные нормально распределены.
  Альтернативная гипотеза (H1): Данные не нормально распределены.

Если p-value меньше уровня значимости (α = 5% или 0.05), то мы отвергаем нулевую гипотезу и считаем, что данные не нормально распределены.

В этом случае:

  p-value = 0.00002 — это значительно меньше 0.05.
  Следовательно, мы отвергаем нулевую гипотезу и делаем вывод, что данные не распределены нормально.

2. Выбор статистического критерия

Поскольку данные не являются нормально распределёнными, лучше использовать непараметрический критерий, который не требует нормальности данных. Подходящие методы для сравнения CPA (стоимость привлечения клиента) в двух группах:
Манна-Уитни (U-критерий)

Это непараметрический тест, который используется для сравнения двух независимых групп, когда данные не являются нормально распределёнными.
Он основан на ранговых данных и проверяет, различаются ли распределения двух групп.
Применим для ситуаций, когда данные не следуют нормальному распределению.

Почему нельзя использовать t-тест?

t-тест для независимых выборок предполагает нормальность распределения данных в каждой группе. В нашем случае это условие не выполнено, поэтому t-тест может дать неверные результаты.

Рекомендация:

Используйте U-критерий Манна-Уитни для проверки гипотезы о различии CPA между группами, так как он подходит для анализа данных с нарушением нормальности распределения.


# Задача 5.
Мы провели АБ-тест на увеличение average timespent per user. По итогам тестирования мы
получили следующие данные. Является ли результат статистически значимым с уровнем
доверия 80%? Какую версию мы выкатим на продакшн?
A) Средняя - 360, отклонение - 40, количество - 9802
B) Средняя - 352, отклонение - 58, количество - 9789

Для того чтобы ответить на вопрос о статистической значимости результатов A/B теста, нам нужно выполнить следующие шаги:
Шаг 1: Определение гипотез

Нулевая гипотеза (H₀): Среднее время, проведённое пользователем на сайте (average timespent per user), в группах A и B не отличается.
Альтернативная гипотеза (H₁): Среднее время, проведённое пользователем на сайте, в группах различается.

Шаг 2: Использование t-теста для двух независимых выборок

Поскольку у нас есть средние, стандартные отклонения и размеры выборок, мы можем использовать t-тест для независимых выборок, чтобы определить, есть ли статистически значимые различия между группами A и B.

Формула t-статистики:

t = XA − XB / SA2nA+SB2nB
t=nA​SA2​​+nB​SB2​​
​XA​ˉ​−XB​ˉ​​

Где:

XAˉXA​ˉ​ и XBˉXB​ˉ​ — средние значения для групп A и B.
SASA​ и SBSB​ — стандартные отклонения для групп A и B.
nAnA​ и nBnB​ — размеры выборок в группах A и B.

Шаг 3: Определение уровня доверия и p-value

Уровень доверия 80% означает, что уровень значимости (α) составляет 20% или 0.20.

Теперь рассчитаем t-статистику и p-value, чтобы определить, есть ли различия.

Похоже, я не могу выполнить более сложный анализ данных прямо сейчас. Пожалуйста, попробуй позже, и я помогу с расчетами.

Ты можешь использовать следующие шаги для проведения t-теста:

Найди t-статистику по формуле:
t=XAˉ−XBˉSA2nA+SB2nB
t=nA​SA2​​+nB​SB2​​

​XA​ˉ​−XB​ˉ​​
Используй таблицу критических значений t-распределения для получения p-value или проведи t-тест с помощью Python:

In [10]:
from scipy import stats
t_stat, p_value = stats.ttest_ind_from_stats(mean1=360, std1=40, nobs1=9802, mean2=352, std2=58, nobs2=9789)


# Задача 6.
Создайте техническую архитектуру проекта по аб тестированию продукта он-лайн кинотеатра
с учетом кросс-девайс аналитики по следующей гипотезе:
Если договориться с банком о 99% кэшбэке на подписку первого месяца, то это повысит
конверсию в подписку на 30%, благодаря упрощенному принятию решения со стороны
пользователя.
На схеме необходимо отобразить:
1) Управленческий процесс по договоренностям с внешними партнерами
2) Архитектуру данных с указанием систем, из которых будем скачивать данные
3) Внутрикомандное взаимодействие

Техническая архитектура проекта по A/B-тестированию для онлайн-кинотеатра с учетом кросс-девайс аналитики:
1. Управленческий процесс по договоренностям с внешними партнерами

    Команда по работе с партнерами:
        - Установление связи с банками.
        - Проведение переговоров о кэшбэке.
        - Заключение соглашений.

    Управление проектом:
        - Назначение ответственных за реализацию теста.
        - Мониторинг выполнения задач.
        - Координация между отделами.

2. Архитектура данных

        - CRM-система: Данные о пользователях, подписках и взаимодействиях.
        - Система аналитики (например, Google Analytics): Поведение пользователей на сайте.
        - Система A/B-тестирования (например, Optimizely или Amplitude):
        - Управление тестами и сбор результатов.
        - База данных: Хранение данных о тестах, результатах и показателях конверсии.
        - Система отчетности (например, Tableau или Power BI): Визуализация и анализ результатов.

3. Внутрикомандное взаимодействие

    Команда разработки:
        - Разработка функционала для внедрения кэшбэка.
        - Обновление интерфейса для пользователей.

    Команда маркетинга:
        - Подготовка рекламной кампании о новом предложении.
        - Сбор обратной связи от пользователей.

    Аналитическая команда:
        - Настройка A/B-тестирования.
        - Сбор и анализ данных, выводы о конверсии.

